src/main/java/com/gluonhq/substrate/model/Triplet.java

```diff
@@ -35,6 +35,7 @@
 
 public class Triplet {
 
+    private final Profile profile;
```

Comment:
At the very least, this should not be final.
I think adding Profile here is dangerous, as Profile is just something to make the triplet human-readable, but build tools should never make decisions based on the single "Profile" info. This is why we have the 3 things that distinguish profiles from each other explicitly as variables here.

Reply:
removed field